File: Backend/StudentManagementSystem/src/main/java/com/ManagementApplication/StudentManagementSystem/service/EnrollmentService.java
package com.ManagementApplication.StudentManagementSystem.service;

import com.ManagementApplication.StudentManagementSystem.dto.EnrollmentDto;
import com.ManagementApplication.StudentManagementSystem.exception.ResourceNotFoundException;
import java.util.List;

public interface EnrollmentService {
    List<EnrollmentDto> getAllEnrollments();

    EnrollmentDto getEnrollmentById(Long id) throws ResourceNotFoundException;

    EnrollmentDto createEnrollment(EnrollmentDto enrollmentDto);

    EnrollmentDto updateEnrollmentById(Long id, EnrollmentDto enrollmentDto) throws ResourceNotFoundException;

    void deleteEnrollmentById(Long id) throws ResourceNotFoundException;

    List<EnrollmentDto> getEnrollmentsByStudentId(Long studentId);

    List<EnrollmentDto> getEnrollmentsByCourseId(Long courseId);
}

File: Frontend/public/src/services/EnrollmentService.js
import axios from "axios";

const ENROLLMENT_API_BASE_URL = 'http://localhost:8080/api/enrollments';

class EnrollmentService {
    getAllEnrollments() {
        return axios.get(ENROLLMENT_API_BASE_URL);
    }

    getEnrollmentById(id) {
        return axios.get(ENROLLMENT_API_BASE_URL + '/' + id);
    }

    createEnrollment(enrollment) {
        return axios.post(ENROLLMENT_API_BASE_URL, enrollment);
    }

    updateEnrollment(id, enrollment) {
        return axios.put(ENROLLMENT_API_BASE_URL + '/' + id, enrollment);
    }

    deleteEnrollment(id) {
        return axios.delete(ENROLLMENT_API_BASE_URL + '/' + id);
    }

    getEnrollmentsByStudentId(studentId) {
        return axios.get(ENROLLMENT_API_BASE_URL + '/student/' + studentId);
    }

    getEnrollmentsByCourseId(courseId) {
        return axios.get(ENROLLMENT_API_BASE_URL + '/course/' + courseId);
    }
}

export default new EnrollmentService(); 